tests/test_cmake_build/embedded.cpp

```diff
@@ -0,0 +1,21 @@
+#include <pybind11/embedded.h>
+namespace py = pybind11;
+
+PYBIND11_ADD_EMBEDDED_MODULE(test_cmake_build)(py::module &m) {
+    m.def("add", [](int i, int j) { return i + j; });
```

Comment:
Regarding the multi-phase initialization: this makes sense, and I agree that it would be desirable to support it (with the appropriate fallbacks for Python 2.7). I find the macro with double parentheses somewhat painful. How about the following?

```
PYBIND11_EMBEDDED_MODULE(name, py::module &m) { ... }
PYBIND11_MODULE(name, py::module &m) { ... }
```
Note also the removal of ``_ADD_`` -- the fact that there can be multiple embedded modules is something that should be explained in the documentation, but I don't think we put the explanation into the macro names.